WalletWasabi/WabiSabi/Client/CoinJoinClient.cs

```diff
@@ -0,0 +1,202 @@
+		private SecureRandom SecureRandom { get; }
+		private CancellationTokenSource DisposeCts { get; } = new();
+		private Coin[] Coins { get; set; }
+		private Random Random { get; } = new();
+
+		public CoinJoinClient(
+			ClientRound round,
+			IArenaRequestHandler arenaRequestHandler,
+			IEnumerable<Coin> coins,
+			Kitchen kitchen,
+			KeyManager keymanager)
+		{
+			Round = round;
+			ArenaRequestHandler = arenaRequestHandler;
+			Kitchen = kitchen;
+			Keymanager = keymanager;
+			SecureRandom = new SecureRandom();
+			Coins = coins.ToArray();
+		}
+
+		protected override async Task ExecuteAsync(CancellationToken stoppingToken)
+		{
+			try
+			{
+				// Register coins.
+				AliceClient[] aliceClients = await RegisterCoinsAsync(stoppingToken).ConfigureAwait(false);
+
+				// Confirm coins.
+				await ConfirmConnectionsAsync(aliceClients, stoppingToken).ConfigureAwait(false);
+
+				// Planning
+				var outputs = DecomposeAmounts(stoppingToken);
+
+				// Output registration.
+				await ReissueAndRegisterOutputsAsync(outputs, stoppingToken).ConfigureAwait(false);
+
+				Transaction? unsignedCoinJoinTransaction = null; // TODO: Get it from somewhere.
+
+				// Send signature.
+				await SignTransactionAsync(aliceClients, unsignedCoinJoinTransaction).ConfigureAwait(false);
+			}
+			catch (Exception ex)
+			{
+				// The game is over for this round, no fallback mechanism. In the next round we will create another CoinJoinClient and try again.
+			}
+		}
+
+		public async Task StartMixingCoinsAsync()
+		{
+			await StartAsync(DisposeCts.Token).ConfigureAwait(false);
+		}
+
+		private async Task<AliceClient[]> RegisterCoinsAsync(CancellationToken stoppingToken)
+		{
+			var aliceArenaClient = new ArenaClient(
+				Round.AmountCredentialIssuerParameters,
+				Round.VsizeCredentialIssuerParameters,
+				AmountCredentialPool,
+				VsizeCredentialPool,
+				ArenaRequestHandler,
+				SecureRandom);
+
+			List<AliceClient> aliceClients = new();
+
+			foreach (var coin in Coins)
+			{
+				var secret = Keymanager.GetSecrets(Kitchen.SaltSoup(), coin.ScriptPubKey.WitHash.ScriptPubKey).First().PrivateKey.GetBitcoinSecret(Keymanager.GetNetwork());
+				aliceClients.Add(await AliceClient.CreateNewAsync(aliceArenaClient, coin, secret, Round.Id, Round.FeeRate).ConfigureAwait(false));
+				await Task.Delay(Random.Next(0, 1000), stoppingToken).ConfigureAwait(false);
+			}
+
+			return aliceClients.ToArray();
+		}
+
+		private async Task ConfirmConnectionsAsync(IEnumerable<AliceClient> aliceClients, CancellationToken stoppingToken)
+		{
+			foreach (var alice in aliceClients)
+			{
+				await alice.ConfirmConnectionAsync(TimeSpan.FromMilliseconds(Random.Next(1000, 5000)), stoppingToken).ConfigureAwait(false);
+				await Task.Delay(Random.Next(0, 1000), stoppingToken).ConfigureAwait(false);
+			}
+		}
+
+		private async Task ReissueAndRegisterOutputsAsync(IEnumerable<(Money Amount, HdPubKey Pubkey)> outputs, CancellationToken stoppingToken)
```

Comment:
After we merge https://github.com/zkSNACKs/WalletWasabi/pull/5689 all this methods will receive a `CancellationToken` that needs to be passed down to all the invoked async methods.

Reply:
Done.